types/topology.go

```diff
@@ -217,6 +217,20 @@ func (t *Topology) GetNodeEnforceStartupConfig(name string) bool {
 	return false
 }
 
+func (t *Topology) GetNodeSuppressStartupConfig(name string) bool {
+	if ndef, ok := t.Nodes[name]; ok {
+		if ndef.GetSuppressStartupConfig() {
+			return true
+		}
+		if t.GetKind(t.GetNodeKind(name)).GetSuppressStartupConfig() {
+			return true
+		}
+		return t.GetDefaults().GetSuppressStartupConfig()
```

Comment:
The issue with this is, that you can not set SuppressStartupConfig on the default or kind level but overwrite it on the more specific level (kind or node) because if you explicitly set it to false, it will evaluate the higher level.
The field must be defined as a pointer  with yaml omitempty hence default nil and not false and the evaluation here needs to get adjusted as well.

Reply:
Thanks for the feedback.  Refactored as a boolean and pushed a new changeset.  Let me know if a third test case is warranted with an explicit false config setting

Reply:
Id make  GetNodeSuppressStartupConfig() return a bool not the pointer because that func is able to resolve the finit value from the hierarchy. But itself it need the info if the setting is defined on a per level basis.

Reply:
That sounds like halfway between the approach of `topology.GetNodeEnforceStartupConfig` and `topology.GetNodeAutoRemove`, without really matching either of them.  Is it worth adding a third method?

Would also imply needing to test `GetNodeSuppressStartupConfig` directly in `config_test.go`, I imagine?

Reply:
pushed new changeset with recommended changes and additional tests

Reply:
You are right. 
In my viewe should also adjust the two functions mentioned.
- `topology.GetNodeEnforceStartupConfig`
- `topology.GetNodeAutoRemove`

to also resolve bool pointers and return actual bools, to figure out if a certain value was set at a given level or not (allow for overwrites). @hellt, whats your opinion?

Reply:
@steiler  yes, I agree, that it would be better to unify the logic for boolean node properties to use pointers as NodeDefinition fields, but return bool when *topology.GetXXX is used

Reply:
alignment via #1567